postgres-types/src/lib.rs

```diff
@@ -951,3 +951,21 @@ fn downcast(len: usize) -> Result<i32, Box<dyn Error + Sync + Send>> {
         Ok(len as i32)
     }
 }
+
+/// A helper trait to be able create a parameters iterator from `&dyn ToSql` or `T: ToSql`
+pub trait BorrowToSql {
```

Comment:
It's probably best to make this opaque and seal it like we do for `RowIndex` just in case.